### PR TITLE
Pro 4588 add user settings module

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*
This PR updates the documentation to include how to use the new settings module. Closes PRO-4588.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Build the documentation
2. Navigate to `/reference/modules/settings.htm`, `/reference/modules/i18n.html`, and `/guide/users.html`

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: docs/guide/users.md [hunk 1]
```diff
@@ -24,16 +24,10 @@ Open the users manager interface by clicking the "Users" button in the admin men
 
 ![The admin menu at the top of the browser window with the Users button highlighted](/images/users-admin-bar.png)
 
-When the manager opens, click the "New User" button to open a fresh content editor modal. Populate the required fields, including the user's "Display Name," [role](#user-roles), username, and password.
+When the manager opens, click the "New User" button to open a fresh content editor modal. Populate the required fields, including the user's "Display Name," [role](#user-roles), username, and password. At the same time, you can set the user admin UI language if the `@apostrophecms/i18n` module is configured for multiple `adminLocales` languages. See the [reference documentation for the `i18n` module](/reference/modules/i18n.html) for more information. If this hasn't been configured, then this selection input won't be shown. Otherwise, it will allow the user to select for the admin UI to match the language for the locale, or remain set to a specific language, no matter the locale language.
```
Comment (boutell) [inline docs/guide/users.md:27]: This is still a little confusing.

How about:

"By default, if a site is configured for multiple locales [link here], and a user is viewing the site in a particular locale, then the user interface will also be displayed in that locale whenever possible. However, if the `adminLocales` option of the `i18n` module is set to an array of locales, then it becomes possible to 'lock' the admin UI to a particular language on a per-user basis. Users can change this setting themselves if the Personal Settings module is configured."

File: docs/reference/modules/i18n.md [hunk 2]
```diff
@@ -122,6 +125,56 @@ Same thing if we request `example.ca` with no locale prefix, we will be redirect
   </template>
 </AposCodeBlock>
 
+### `adminLocales` {#adminlocales}
+
+The `adminLocales` takes an array of objects. Each of the objects should have a label that is presented to the user in both the user manager editor, and in the user preferences menu. This menu can be accessed through the dropdown at the right hand side of the admin-bar menu. The objects should also have a `value` property that takes the key for a locale defined in the `locales` option, that should be associated with this selection.
```
Comment (boutell) [inline docs/reference/modules/i18n.md:130]: There is no requirement for adminLocales to also appear under locales. If we have translations for a locale, or you bring them at project level, you can specify it for adminLocales even if you have no intention of offering content in that locale. This is fairly common ("this site is in three non-English languages but we want the admin UI in business English").

File: docs/reference/modules/settings.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+extends: '@apostrophecms/module'
+---
+
+# `@apostrophecms/settings`
+
+**Alias:** `apos.settings`
+
+<AposRefExtends :module="$frontmatter.extends" />
+
+This module governs the Personal Settings menu that allows users to change their preferences. Apostrophe projects can configure this menu through `@apostrophecms/settings`.
```
Comment (boutell) [inline docs/reference/modules/settings.md:11]: As such, these settings are **personal** to the user. If you're looking for sitewide settings that govern overall site behavior, check out `@apostrophecms/global`.

File: docs/reference/modules/settings.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+## Options
+
+|  Property | Type | Description |
+|---|---|---|
+| [`subforms`](#subforms) | Object | Each object defines an input field in the Personal Settings menu. |
```
Comment (boutell) [inline docs/reference/modules/settings.md:17]: a set of input fields in the Personal Settings dialog box, with its own Update button.

File: docs/reference/modules/settings.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+| [`groups`](#groups) | Object | Like the property of the same name in the `fields` setting, this option organizes the subforms on tabs in the Personal Settings menu. |
```
Comment (boutell) [inline docs/reference/modules/settings.md:18]: Use of the word `setting` for anything else in this article is probably too confusing.

File: docs/reference/modules/settings.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+In addition to the required `fields` property, each object can take five additional optional properties.
+
+The `label` property takes a string to display to the user in both preview and edit mode. If not supplied, Apostrophe will use the label for the first schema passed to the `fields` property. 
+
+The `help` property takes a field to display to the user in preview mode. If this property is present, it will replace any component added through the option `previewComponent` property. The `previewComponent` property takes the name of a Vue component that has been added to your project through any `modules/my-custom-module/ui/apos/components` folder. The string passed to this property should not contain the file extension. [See below](#previewcomponent) for further information about the component structure.
+
+The `protection` property can take a value of either `true` or `"password"`. These values are equivalent and will require that the user enters their password when attempting to change the value(s) of this `fields` object. Additional fields that are added by default can be added through the `addProtectedField()` method of the module.
```
Comment (boutell) [inline docs/reference/modules/settings.md:76]: Equivalent for now, yeah. There will be an option to change the behavior of `true` later, for SSO based sites where `password` isn't useful.

File: docs/reference/modules/settings.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+Finally, the `reload` property defaults to false, but can take a value of `true` if the preview should update after the user makes a change. This isn't needed if the `help` property is set.
```
Comment (boutell) [inline docs/reference/modules/settings.md:78]: ... Can take a value of `true` if the page should refresh after the user makes a change. We want the user to understand that the setting took effect, so this is useful any time a personal setting ought to have an immediate impact on the page, e.g. displaying their name on a menu, etc.

File: docs/guide/users.md (hunk 1 above)
Comment (boutell) [reply docs/guide/users.md:27]: Also see `defaultAdminLocale` which needs to be covered here.

File: docs/reference/modules/settings.md [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+In addition to the required `fields` property, each object can take five additional optional properties.
+
+The `label` property takes a string to display to the user in both preview and edit mode. If not supplied, Apostrophe will use the label for the first schema passed to the `fields` property. 
+
+The `help` property takes a field to display to the user in preview mode. If this property is present, it will replace any component added through the option `previewComponent` property. The `previewComponent` property takes the name of a Vue component that has been added to your project through any `modules/my-custom-module/ui/apos/components` folder. The string passed to this property should not contain the file extension. [See below](#previewcomponent) for further information about the component structure.
```
Comment (boutell) [inline docs/reference/modules/settings.md:74]: "Help" is text (or an i18n key) that appears when in preview mode (when not editing that subform). If it is not present then the i18n key "preview" is used, and this is typically an i18next key that receives placeholders like `{{ firstName }} {{ lastName}}`, leveraging i18next as a built-in templating system and allowing different locales to put them in different orders as appropriate. If neither is present then `previewComponent` is rendered via Vue. If `previewComponent` doesn't exist either, then the fields of the subform are concatenated as space-separated strings for preview purposes.

TODO: I'm not entirely sure why both `help` and `preview` need to exist and I've asked Miro.

File: docs/reference/modules/settings.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+### `previewComponent`
+
+The preview component is displayed to the right of a `subforms` label when the user first opens a tab. If not supplied, Apostrophe will attempt to create a preview by combining the values of any schema fields within the individual `subforms` object as a space-separated string. For example, if the object has two schema fields, `firstName` and `lastName`, the resulting preview would be the equivalent to a template with `{{ firstName }} {{ lastName }}`. While this would work well for an object with a small number of fields, this might not be optimal for an object with a larger number.
```
Comment (boutell) [inline docs/reference/modules/settings.md:86]: Not quite right, it's `previewComponent` or else `preview` or else the automatic behavior you describe (and `help` plays some role which I have asked Miro to clarify).

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/guide/users.md (hunk 1 above)
Comment (BoDonkey) [reply docs/guide/users.md:27]: Changed.

File: docs/reference/modules/i18n.md (hunk 2 above)
Comment (BoDonkey) [reply docs/reference/modules/i18n.md:130]: Changed.

File: docs/reference/modules/settings.md (hunk 3 above)
Comment (BoDonkey) [reply docs/reference/modules/settings.md:11]: Changed.

File: docs/reference/modules/settings.md (hunk 5 above)
Comment (BoDonkey) [reply docs/reference/modules/settings.md:18]: Changed.

File: docs/reference/modules/settings.md (hunk 7 above)
Comment (BoDonkey) [reply docs/reference/modules/settings.md:78]: Changed, not sure how much of the comment was for me and how much was for the actual documentation.

File: docs/reference/modules/settings.md (hunk 9 above)
Comment (BoDonkey) [reply docs/reference/modules/settings.md:86]: Changed.

File: docs/reference/modules/settings.md (hunk 8 above)
Comment (BoDonkey) [reply docs/reference/modules/settings.md]: I rewrote/rearranged this after talking with Miro a bit. `previewComponent`>`help`>`preview`>automatic assignment

**Review by boutell — APPROVED** [review]
Approving but I also have a query out to Miro for more info about help vs preview for followup if we are able to clarify.

**Review by boutell — APPROVED** [review]

**Pull request merged**